Expensify/App
# IOU - Localize amount text input

### Details

IOU - Localize amount text input. Users will be able to type/paste localized amount format in amount input. See also: https://github.com/Expensify/App/issues/6427#issuecomment-1002950968

### Fixed Issues

$ https://github.com/Expensify/App/issues/6427

### Tests


1. Go to preference set language to "Español"
2. Request money from chats list
3. Verify that the virtual keypad on mobile shows localized digits.
4. Verify that localized amount text can be typed with virtual keypad on mobile.
5. Verify that localized amount text can be typed with hardware keyboard on desktop.
6. Verify that localized amount text can be pasted.

### QA Steps

Same as above.

### Tested On


### Screenshots


#### Web

<img width="1100" alt="web-1" src="https://user-images.githubusercontent.com/19236594/149615758-636c4729-38d5-42a4-9831-d77525647b35.png">
<img width="1100" alt="web-2" src="https://user-images.githubusercontent.com/19236594/149615759-30c2c16b-544e-402d-998f-301039852eec.png">
<img width="1100" alt="web-3" src="https://user-images.githubusercontent.com/19236594/149615760-456eff30-4f77-4ad3-80bf-2bcdd36aef5c.png">

#### Mobile Web

<img width="353" alt="mobile-web-1" src="https://user-images.githubusercontent.com/19236594/149615792-5cc7ca95-913d-47b0-9f89-7b6f6dfc1ba9.png">
<img width="353" alt="mobile-web-2" src="https://user-images.githubusercontent.com/19236594/149615795-9437f942-9fb3-4322-a34c-c7526b1a5a33.png">
<img width="353" alt="mobile-web-3" src="https://user-images.githubusercontent.com/19236594/149615796-9778fc78-f3fb-43ff-ad29-1bcf4f2ca7bd.png">

#### Desktop

<img width="1113" alt="desktop-1" src="https://user-images.githubusercontent.com/19236594/149615803-745416c7-9806-47a8-a85a-cfeba03b22cd.png">
<img width="1113" alt="desktop-2" src="https://user-images.githubusercontent.com/19236594/149615806-94982325-7d3d-487d-84b0-571b553c56b3.png">
<img width="1113" alt="desktop-3" src="https://user-images.githubusercontent.com/19236594/149615808-3b7237c1-4e7c-4e3f-b023-9c8e161e9723.png">

#### iOS

<img width="353" alt="ios-1" src="https://user-images.githubusercontent.com/19236594/149615810-99624791-4de9-4706-8e02-0385532439ff.png">
<img width="353" alt="ios-2" src="https://user-images.githubusercontent.com/19236594/149615813-f1053136-f163-4597-820b-6c59da725867.png">
<img width="353" alt="ios-3" src="https://user-images.githubusercontent.com/19236594/149615814-3193be5b-cf85-4fc9-a0e5-94d3cd574004.png">

#### Android

<img width="353" alt="android-1" src="https://user-images.githubusercontent.com/19236594/149615820-f634cdeb-fe61-47fa-b27f-a5ca61a9a065.jpg">
<img width="353" alt="android-2" src="https://user-images.githubusercontent.com/19236594/149615822-f2b0a40b-a8c9-4c1f-baa5-6c7fc767eb22.jpg">
<img width="353" alt="android-3" src="https://user-images.githubusercontent.com/19236594/149615823-3877d60a-9788-4c18-9ad8-ce0535409121.jpg">
